packages/replay-next/src/contexts/FocusContext.tsx

```diff
@@ -81,7 +84,7 @@ export function FocusContextRoot({ children }: PropsWithChildren<{}>) {
 
   // Refine the loaded ranges based on the focus window.
   useEffect(() => {
-    if (range === null) {
+    if (range === null || range === initialRange) {
```

Comment:
I think this might be a bug waiting to happen if we:
1. Start with the initial range
2. Change it to something else
3. Change it back to the full range using the focuser UI

Wouldn't we end up _not_ calling `client.requestFocusRange` at step 3 even though we should?

I think what you're trying to do is avoid calling the API if the range hasn't changed from whatever it started as. I think we can do that using a ref though?

```js
const prevRangeRef = useRef(initialRange);

useEffect(() => {
  if (range !== prevRangeRef.current) {
    prevRangeRef.current = range;
```

The ref would be initialized the first time the component renders (so we wouldn't request initially) and then we'd request each time it changed afterward.

Reply:
> Wouldn't we end up not calling client.requestFocusRange at step 3 even though we should?

No, because the `range` will still not be the same *object* even if it contains the same values.

Reply:
Ah, right, because you wrapped it with `useMemo` 🤔 I guess that wouldn't break but it's also not the clearest relationship.